src/Utils/PlatformFeatureUtils.ts

```diff
@@ -0,0 +1,170 @@
+    PlatformFeature.ReadDocument,
+    PlatformFeature.UpdateDocument,
+    PlatformFeature.DeleteDocument,
+  ]),
+
+  // Deprecate after we remove Platform.Fabric dependency, it should always be FabricNative or FabricMirror
+  [Platform.Fabric]: new Set([
+    PlatformFeature.Queries,
+    PlatformFeature.EnhancedQueryControl,
+    PlatformFeature.VectorSearch,
+    PlatformFeature.FullTextSearch,
+    PlatformFeature.ComputedProperties,
+
+    PlatformFeature.ReadDatabase,
+    PlatformFeature.DeleteDatabase,
+
+    PlatformFeature.CreateCollection,
+    PlatformFeature.ReadCollection,
+    PlatformFeature.DeleteCollection,
+
+    PlatformFeature.CreateDocument,
+    PlatformFeature.ReadDocument,
+    PlatformFeature.UpdateDocument,
+    PlatformFeature.DeleteDocument,
+  ]),
+
+  [Platform.FabricNative]: new Set([
+    PlatformFeature.Queries,
+    PlatformFeature.EnhancedQueryControl,
+    PlatformFeature.VectorSearch,
+    PlatformFeature.FullTextSearch,
+    PlatformFeature.ComputedProperties,
+
+    PlatformFeature.CreateDatabase,
+    PlatformFeature.ReadDatabase,
+    PlatformFeature.DeleteDatabase,
+
+    PlatformFeature.CreateCollection,
+    PlatformFeature.ReadCollection,
+    PlatformFeature.DeleteCollection,
+
+    PlatformFeature.CreateDocument,
+    PlatformFeature.ReadDocument,
+    PlatformFeature.UpdateDocument,
+    PlatformFeature.DeleteDocument,
+  ]),
+
+  [Platform.FabricMirrored]: new Set([
+    PlatformFeature.Queries,
+    PlatformFeature.EnhancedQueryControl,
+    PlatformFeature.VectorSearch,
+    PlatformFeature.FullTextSearch,
+    PlatformFeature.ComputedProperties,
+
+    PlatformFeature.CreateDatabase,
+    PlatformFeature.ReadDatabase,
+    PlatformFeature.DeleteDatabase,
+
+    PlatformFeature.CreateCollection,
+    PlatformFeature.ReadCollection,
+    PlatformFeature.DeleteCollection,
+
+    PlatformFeature.CreateDocument,
+    PlatformFeature.ReadDocument,
+    PlatformFeature.UpdateDocument,
+    PlatformFeature.DeleteDocument,
+  ]),
+};
+
+/**
+ * Central feature flag function - checks if a feature is enabled for current platform
+ * @param feature The feature to check
+ * @param platform Optional platform override, defaults to current platform
+ * @returns True if the feature is enabled for the platform, false otherwise
+ */
+export const isFeatureSupported = (feature: PlatformFeature, platform?: Platform): boolean => {
+  const currentPlatform = platform ?? configContext.platform;
+
+  // Portal and Hosted platforms have all features enabled by default
+  if (currentPlatform === Platform.Portal || currentPlatform === Platform.Hosted) {
+    return true;
+  }
+
+  return PLATFORM_FEATURE_MATRIX[currentPlatform]?.has(feature) ?? false;
+};
+
+export const areScriptsSupported = (platform?: Platform): boolean => {
+  const currentPlatform = platform ?? configContext.platform;
+  if (currentPlatform === Platform.Portal || currentPlatform === Platform.Hosted) {
+    return true;
+  }
```

Comment:
Why not remove this and add the 3 script features to each Portal and Hosted. 
This way we have only one source of truth where the features are defined and no additional special logic that turns on/off features.